src/data/monoid_algebra.lean

```diff
@@ -225,10 +225,10 @@ instance [ring k] [monoid G] : ring (monoid_algebra k G) :=
 instance [comm_ring k] [comm_monoid G] : comm_ring (monoid_algebra k G) :=
 { mul_comm := mul_comm, .. monoid_algebra.ring}
 
-instance [semiring k] : has_scalar k (monoid_algebra k G) :=
+instance {R : Type*} [semiring R] [semiring k] [semimodule R k] : has_scalar R (monoid_algebra k G) :=
```

Comment:
Can you switch the names `R` and `k` in this instance and the next one? This way around is too confusing.

Reply:
#4315

Reply:
I've updated #4315 to use a clearer naming